### PR TITLE
Improve guidance when there are no pipelines to show

### PR DESCRIPTION
**Pull request by jwforres**
Fixes https://github.com/openshift/origin-web-console/issues/932

### CODE REVIEW

**Comment by jwforres** [comment]
no-pipelines

**Comment by jwforres** [comment]
@danmcp @spadgett PTAL

**Comment by jwforres** [comment]
the deploy button shows up if we are able to find the sample pipeline template, which is configurable through our extension Constants. It goes straight to the template deploying page for the sample.

**Comment by jwforres** [comment]
Tweaked the spacing within the paragraph

no-pipeline-updated

**Comment by danmcp** [comment]
Awesome!

File: app/scripts/controllers/pipelines.js [hunk 1]
```diff
@@ -106,11 +107,21 @@ angular.module('openshiftConsole')
...
           // Filter on the client until the server supports fieldSelector on spec.strategy.type.
           // Use _.pick instead of _.filter to keep $scope.buildConfigs a map
           $scope.buildConfigs = _.pick(buildConfigData.by("metadata.name"), isPipeline);
+          if (_.isEmpty($scope.buildConfigs) && !checkedForSampleTemplate) {
+            checkedForSampleTemplate = true;
+            var sampleName = Constants.SAMPLE_PIPELINE_TEMPLATE.name;
+            var sampleNamespace = Constants.SAMPLE_PIPELINE_TEMPLATE.namespace;
+            DataService.get("templates", sampleName, {namespace: sampleNamespace}, {errorNotification: false}).then(
+              function() {
+                $scope.deploySampleURL = "project/" + $scope.projectName + "/create/fromtemplate?name=" + encodeURIComponent(sampleName) + "&namespace=" + encodeURIComponent(sampleNamespace);
```
Comment (spadgett) [inline app/scripts/controllers/pipelines.js]: We should use `Navigate.fromTemplateURL`. Then this won't need to change when the parameter names change in #884

File: app/views/pipelines.html [hunk 2]
```diff
@@ -27,10 +27,15 @@
...
+                    <a ng-href="{{ 'pipeline-builds' | helpLink}}" target="_blank">Pipeline Builds</a>
+                    and the
+                    <a ng-href="{{ 'pipeline-plugin' | helpLink}}" target="_blank">OpenShift Pipeline Plugin</a>.
                   </p>
-                  <p ng-if="project.metadata.name | canIAddToProject">
-                    <a ng-href="project/{{projectName}}/create" class="btn btn-lg btn-primary">
-                      Add to Project
+                  <p ng-if="(project.metadata.name | canIAddToProject) && deploySampleURL">
+                    <a ng-href="{{deploySampleURL}}" class="btn btn-lg btn-primary">
+                      Deploy Sample Pipeline
```
Comment (spadgett) [inline app/views/pipelines.html]: Is "Create Sample Pipeline" better?

**Comment by jwforres** [comment]
updated based on feedback [merge]

**Pull request merged**